### PR TITLE
MINOR: Scala cleanups in core

### PR DESCRIPTION
**Pull request by mimaison**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/server/DelegationTokenManager.scala [hunk 1]
```diff
@@ -140,13 +140,13 @@ object DelegationTokenManager {
 
     val allow =
     //exclude tokens which are not requested
-      if (!owners.isEmpty && !owners.get.exists(owner => token.ownerOrRenewer(owner))) {
+      if (owners.isDefined && !owners.get.exists(owner => token.ownerOrRenewer(owner))) {
```
Comment (divijvaidya) [inline core/src/main/scala/kafka/server/DelegationTokenManager.scala:143]: although `isDefined` and `nonEmpty` are equivalent but I was wondering if there is a reason for using former at some places and the latter at others?

**Review by divijvaidya — APPROVED** [review]
Looks good to me.

I have done a similar clean up at https://github.com/apache/kafka/pull/12004. Please take a look when you get a chance.

Thank you.

File: core/src/main/scala/kafka/server/DelegationTokenManager.scala (hunk 1 above)
Comment (mimaison) [reply core/src/main/scala/kafka/server/DelegationTokenManager.scala:143]: `isDefined` and `nonEmpty` are equivalent on `Option`. On `String` there is not `isDefined`, so we need to use `nonEmpty`

File: core/src/main/scala/kafka/server/PartitionMetadataFile.scala [hunk 2]
```diff
@@ -44,8 +44,7 @@ object PartitionMetadataFile {
   }
 
   class PartitionMetadataReadBuffer[T](location: String,
-                                       reader: BufferedReader,
-                                       version: Int) extends Logging {
+                                       reader: BufferedReader) extends Logging {
```
Comment (showuon) [inline core/src/main/scala/kafka/server/PartitionMetadataFile.scala:47]: nice cleanup for unused argument

**Review by showuon — COMMENTED** [review]
@mimaison , thanks for the PR. Mostly LGTM! The only thing I'm not quite sure if why we remove `Unit` return type? I know this can be skipped, but I saw there are many other places still keeping the `Unit` type. I'm just not sure if it's necessary.

**Comment by mimaison** [comment]
I've merged https://github.com/apache/kafka/pull/12004 and rebased this on top.

@showuon This PR adds `Unit` wherever needed.

**Review by showuon — APPROVED** [review]

**Comment by showuon** [comment]
> @showuon This PR adds Unit wherever needed.

Ah, you're right! I look at the wrong way. LGTM, then! Thanks.

**Pull request merged**